src/PrinterLog.php

```diff
@@ -121,7 +121,22 @@ public function getMetrics(Printer $printer, $user_filters = []): array
             ] + $filters
         ]);
 
-        return iterator_to_array($iterator);
+        $series = iterator_to_array($iterator);
+
+        // Reduce the data to 25 points
+        $count = count($series);
+        $max_size = 25;
+        if ($count > $max_size) {
+            // Keep one row every X entry using modulo
+            $modulo = round($count / $max_size);
+            $series = array_filter(
+                $series,
+                fn($k) => ($k % $modulo) == 0,
+                ARRAY_FILTER_USE_KEY
+            );
+        }
```

Comment:
I think we should ensure that the last entry is always displayed. For instance, if you have 51 entries in the serie, modulo will be 2, and so the last entry will not be displayed.

Also, using round and modulo will not always limit displayed points count to 25 :
- with 37 entries, 37 points will be displayed (`round(37/25) => 1`);
- with 38 entries, 19 points will be displayed (`round(38/25) => 2`);
- with 62 entries, 31 points will be displayed (`round(62/25) => 2`);
- with 63 entries, 21 points will be displayed (`round(63/25) => 3`);

Reply:
Yeah, this is kind of hacky, feel free to improve it if you can.
Tbh, it feels bad spending time improving this when the graph lib that should handle this itself by picking its data points from the set.
I know @orthagh is trying a PoC to switch to echarts, which would support this feature.